automata/fa/dfa.py

```diff
@@ -222,6 +226,102 @@ def __len__(self) -> int:
         """Returns the cardinality of the language represented by the DFA."""
         return self.cardinality()
 
+    def to_partial(self) -> Self:
```

Comment:
@EduardoGoulart1 @eliotwrobson This new method needs to be added to the API Documentation (in `docs/fa/class-dfa.md`).

Reply:
@caleb531 updated here!

Reply:
@eliotwrobson Yep, just reviewed it! Looks great to me!